tls/s2n_client_hello.c

```diff
@@ -562,3 +561,19 @@ ssize_t s2n_client_hello_get_extension_by_id(struct s2n_client_hello *ch, s2n_tl
     POSIX_CHECKED_MEMCPY(out, parsed_extension->extension.data, len);
     return len;
 }
+
+ssize_t s2n_client_hello_get_session_id_length(struct s2n_client_hello *ch)
+{
+    POSIX_ENSURE_REF(ch);
+    return ch->session_id.size;
+}
```

Comment:
Can you make this a separate argument, so that the return value only indicates success/failure? Same with the other method.

Reply:
Definitely. If we could go back and change the other APIs we would. But any new ones moving forward should use `out` params instead.

Reply:
Sure, sounds good.